src/ZoneServer/World/Maps/Map.cs

```diff
@@ -190,8 +191,23 @@ private void Disappearances()
 			lock (_monsters)
 				toDisappear = _monsters.Values.Where(a => a.DisappearTime < now).ToList();
 
-			foreach (var monster in toDisappear)
+			foreach (var monster in toDisappear) 
+			{
+
+				// If this is a blue orb summon, remove the variable from the summoner
+				if (monster is Mob mob) 
+				{
+					if (mob.MonsterType != MonsterType.Mob && mob.Components.Get<AiComponent>()?.Script.GetMaster() is Character master)
+					{
+						master.Variables.Temp.Remove("Melia.BlueOrbSummon.Monster");
```

Comment:
Not a big fan of this happening in multiple places, as the redundancy makes it more likely that one instance might be changed but not the other. At the very least this should probably be a method, perhaps on Mob. Though it would be even better if we didn't need to explicitly add it in multiple places in the first place.

Hm, I just realized, there is a flaw that needs to be fixed, isn't there? The disappear time is currently two seconds for monsters, which means that a player could summon a new monster in between the kill and the disappearance. The variables would then be reset despite them having a new follower already.

Reply:
I made a method that does this.  As for this particular call, this is actually not intended to operate on dead monsters, but rather ones that expire due to the disappear time limit, as such I added a check that makes sure the monster is still alive at the time of removal that fixes that potential issue.

In theory, we could remove the check on death and only check it here, but this introduces a very minor issue where if you leave the map within 2 seconds of your follower dying, you can save it, since that code won't have executed yet.  As such, I've left the check on death for now.